code/backend/batch/utilities/search/search_handler_base.py

```diff
@@ -17,10 +17,16 @@ def search_with_facets(self, query: str, facets: list[str]):
             return None
         return self.search_client.search(query, facets=facets)
 
-    def get_unique_files(self, results, facet_key: str):
-        if results:
-            return [facet["value"] for facet in results.get_facets()[facet_key]]
-        return []
+    def get_unique_files(self, search_text, field_name):
+        unique_files = set()  # Using a set to automatically handle uniqueness
+        results = self.search_client.search(search_text=search_text)
```

Comment:
Didn't using facets handle the uniqness before?

Reply:
No, problem was not with the uniqueness. Problem was with the facets count which has a default limit of the top 10 facets. Now we have added one extra parameter for facet count in search_with_facets function which will help us to retrieve all the documents from search client.